cmd/alertmanager/main.go

```diff
@@ -317,9 +323,28 @@ func run() int {
 		go peer.Settle(ctx, *gossipInterval*10)
 	}
 
-	alerts, err := mem.NewAlerts(context.Background(), marker, *alertGCInterval, logger)
-	if err != nil {
-		level.Error(logger).Log("err", err)
+	var alerts provider.Alerts
+	if *alertStorageProvider == "mem" {
+		alerts, err = mem.NewAlerts(context.Background(), marker, *alertGCInterval, logger)
+		if err != nil {
+			level.Error(logger).Log("err", err)
+			return 1
+		}
+	} else if *alertStorageProvider == "etcd" {
```

Comment:
MAY: Remove else if logic and replace it with early returns. This may mean you need to move the defer close up.

related: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Reply:
Going to leave it.  Would work for first statement.  Second has casting issues and that is why I kept etcdAlerts separate.